debezium-core/src/main/java/io/debezium/transforms/debeziumtimestampcoverter/DebeziumTimestampConverter.java

```diff
@@ -0,0 +1,698 @@
+                result.set(Calendar.MILLISECOND, origCalendar.get(Calendar.MILLISECOND));
+                return result.getTime();
+            }
+
+            @Override
+            public Schema optionalSchema() {
+                return SchemaBuilder.int32()
+                        .name(org.apache.kafka.connect.data.Time.LOGICAL_NAME)
+                        .version(1).optional().build().schema();
+            }
+        });
+
+        TRANSLATORS.put(TYPE_TIMESTAMP, new TimestampTranslator() {
+            @Override
+            public Date toRaw(Config config, Object orig) {
+                if (!(orig instanceof Date))
+                    throw new DataException("Expected Timestamp to be a java.util.Date, but found " + orig.getClass());
+                return (Date) orig;
+            }
+
+            @Override
+            public Schema typeSchema() {
+                return Timestamp.SCHEMA;
+            }
+
+            @Override
+            public Date toType(Config config, Date orig, String format) {
+                return orig;
+            }
+
+            @Override
+            public Schema optionalSchema() {
+                return SchemaBuilder.int64()
+                        .name(org.apache.kafka.connect.data.Timestamp.LOGICAL_NAME)
+                        .version(1).optional().build().schema();
+            }
+        });
+
+        TRANSLATORS.put(TYPE_EPCOH_DATE, new TimestampTranslator() {
+            @Override
+            public Date toRaw(Config config, Object orig) {
+                return Date.from(java.time.LocalDate.ofEpochDay((Integer) orig).atStartOfDay(ZoneId.systemDefault()).toInstant());
+            }
+
+            @Override
+            public Schema typeSchema() {
+                return org.apache.kafka.connect.data.Date.SCHEMA;
+            }
+
+            @Override
+            public Object toType(Config config, Date orig, String format) {
+                Calendar origCalendar = Calendar.getInstance(UTC);
+                origCalendar.setTime(orig);
+                Calendar result = Calendar.getInstance(UTC);
+                result.setTimeInMillis(0L);
+                result.set(Calendar.HOUR_OF_DAY, origCalendar.get(Calendar.HOUR_OF_DAY));
+                result.set(Calendar.MINUTE, origCalendar.get(Calendar.MINUTE));
+                result.set(Calendar.SECOND, origCalendar.get(Calendar.SECOND));
+                result.set(Calendar.MILLISECOND, origCalendar.get(Calendar.MILLISECOND));
+                return result.getTime();
+            }
+
+            @Override
+            public Schema optionalSchema() {
+                return SchemaBuilder.int32()
+                        .name(org.apache.kafka.connect.data.Date.LOGICAL_NAME)
+                        .version(1).optional().build().schema();
+            }
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_INT64, () -> {
+            return Schema.OPTIONAL_INT64_SCHEMA.schema();
+        });
```

Comment:
If you change `OPTIONAL_TRANSLATORS` to be `Map<String, Schema>`, then each of these can be replaced with:

```
OPTIONAL_TRANSLATORS.put(TYPE_INT64, Schema.OPTIONAL_INT64_SCHEMA.schema());
```
Or, if you really need to register a supplier of `Schema` by having the `OPTIONAL_TRANSLATORS` to be `Map<String, Supplier<Schema>>`, these lines become:

```
OPTIONAL_TRANSLATORS.put(TYPE_INT64, Schema.OPTIONAL_INT64_SCHEMA::schema);
```
That's actually storing a _reference to the `Schema.OPTIONAL_INT64_SCHEMA.schema()` method_ as the value in the map.